From 403e528a2230afe6e0c0b738ab74f35f2f55112c Mon Sep 17 00:00:00 2001
From: Caesar Wang <wxt@rock-chips.com>
Date: Sat, 16 Jun 2018 11:04:25 +0800
Subject: [PATCH] drm/rockchip: vop: the display be turn off when wakeup

As the vop will turn off the all planes when the system wakeup, that can
avoid the splash screen issues.

But the normal display will be show util the
application is refreshed. That's clearly not sensible.

For now, we can stop dma for turning off vop data.

Change-Id: Ieff151bee7813b978bc3cb65f0457e37dc20bf2c
Signed-off-by: Caesar Wang <wxt@rock-chips.com>
---
 drivers/gpu/drm/rockchip/rockchip_drm_vop.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/rockchip/rockchip_drm_vop.c b/drivers/gpu/drm/rockchip/rockchip_drm_vop.c
index 435eda863692..71cd1a6ea907 100644
--- a/drivers/gpu/drm/rockchip/rockchip_drm_vop.c
+++ b/drivers/gpu/drm/rockchip/rockchip_drm_vop.c
@@ -3337,7 +3337,7 @@ static void vop_crtc_atomic_flush(struct drm_crtc *crtc,
 		int ret;
 
 		if (vop->mode_update)
-			vop_disable_all_planes(vop);
+			VOP_CTRL_SET(vop, dma_stop, 1);
 
 		need_wait_vblank = !vop_is_allwin_disabled(vop);
 		if (vop->mode_update && need_wait_vblank)
-- 
2.35.3

